### PR TITLE
N7S AUR#209 reset reason

### PR DESCRIPTION
**Pull request by Lurkerpas**

### CODE REVIEW

File: requirements/AUR_RT-0040/reset-reason/test.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+echo -e "Building binary\n"
+make leon3 debug
+echo -e "\nExecuting binary\n"
+sis-2.29 -leon3 -dumbio -uart1 system.log -rt work/binaries/leon3 < gdb.batch
+echo -e "\nTest result\n"
+cat system.log
```
Comment (Ficzer) [inline requirements/AUR_RT-0040/reset-reason/test.sh]: Maybe we can place this script in Makefile, one file less

File: requirements/AUR_RT-0040/reset-reason/work/controller/C/src/controller.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    !! file. The up-to-date signatures can be found in the .ads file.   !!
+*/
+#include "controller.h"
+
+#include <CbiPartitionApi.h>
+
+// Implemented in manager.c
+extern void print_rtems(const char *str);
+
+static void startup_on_processor_reset()
```
Comment (Ficzer) [inline requirements/AUR_RT-0040/reset-reason/work/controller/C/src/controller.c:17]: Are one line functions necessary?

File: requirements/AUR_RT-0040/reset-reason/work/controller/C/src/controller.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+{
+   print_rtems("Controller starting up on processor reset\n");
+}
+
+static void startup_on_power_reset()
+{
+   print_rtems("Controller starting up on power reset\n");
```
Comment (Ficzer) [inline requirements/AUR_RT-0040/reset-reason/work/controller/C/src/controller.c]: Maybe "power reset" -> "power on", I think it would be cleaner

**Review by Ficzer — COMMENTED** [review]
These comments apply to all projects

File: requirements/AUR_RT-0040/reset-reason/work/controller/C/src/controller.c (hunk 3 above)
Comment (Lurkerpas) [reply requirements/AUR_RT-0040/reset-reason/work/controller/C/src/controller.c]: Yes, power-on reset is the exact wording of the requirements, will change.

File: requirements/AUR_RT-0040/reset-reason/work/controller/C/src/controller.c (hunk 2 above)
Comment (Lurkerpas) [reply requirements/AUR_RT-0040/reset-reason/work/controller/C/src/controller.c:17]: They are not necessary, but I find them to be more expressive with regards to the requirement fulfillment - think of them as "dedicated startup procedures (named, discrete), filled with dummy implementation (one print)".

File: requirements/AUR_RT-0040/reset-reason/test.sh (hunk 1 above)
Comment (Lurkerpas) [reply requirements/AUR_RT-0040/reset-reason/test.sh]: Makes sense, will do.

**Review by Ficzer — APPROVED** [review]

**Pull request merged**